src/main/java/com/avaje/ebeanservice/elastic/query/EQuery.java

```diff
@@ -39,7 +39,7 @@ public EQuery(BeanType<T> beanType, JsonContext jsonContext, JsonReadOptions opt
    * Create a bean parser for the given json.
    */
   protected BeanSearchParser<T> createParser(JsonParser json) {
-    JsonBeanReader reader = createReader(json);
+    JsonBeanReader<T> reader = createReader(json);
```

Comment:
This one and the ones below were possible because the docstore carries the type along with it and so does the query in several instances.